src/cf.data.pre

```diff
@@ -2952,6 +2952,15 @@ DOC_START
 
 DOC_END
 
+NAME: transparent_outgoing_ip
+TYPE: onoff
+LOC: Config.onoff.transparent_outgoing_ip
+DEFAULT: off
+DOC_START
+	If this is enabled, the outgoing IP address will match the
+	local IP address that the request came in through.
```

Comment:
IMO, this feature should be refactored and reimplemented as a special `tcp_outgoing_address` keyword. For example, the following configuration sketch uses `client-connection-destination-ip` keyword (but I do _not_ insist on that specific keyword spelling and foresee a few other keywords that will be eventually added):

```
tcp_outgoing_address client-connection-destination-ip !localhost
```

I cannot do the required refactoring legwork right now, but I can help with suggestions if you are willing to work on it. Others may be able to offer more help.

Otherwise, the feature is just too limited (i.e. addressing too few/narrow use cases) and too invasive (goes into the middle of a fairly complex and often misunderstood decision making logic) to be officially accepted when there is a proper way to extend an existing directive to cover the new use case.

There are also problems with the feature name and documentation, but I am ignoring them in hope that they will be automatically resolved (or seriously change) during the requested refactoring.